gmp/gmp.py

```diff
@@ -1269,10 +1271,136 @@ def modifyTagCommand(self, tag_id, kwargs):
+            hosts = '<hosts>%s</hosts>' % hosts
+
+        comment = kwargs.get('comment', '')
+        if comment:
+            comment = '<comment>%s</comment>' % comment
+
+        copy = kwargs.get('copy', '')
+        if copy:
+            copy = '<copy>%s</copy>' % copy
+
+        exclude_hosts = kwargs.get('exclude_hosts', '')
+        if exclude_hosts:
+            exclude_hosts = '<exclude_hosts>%s</exclude_hosts>' % exclude_hosts
+
+        alive_tests = kwargs.get('alive_tests', '')
+        if alive_tests:
+            alive_tests = '<alive_tests>%s</alive_tests>' % alive_tests
+
+        reverse_lookup_only = kwargs.get('reverse_lookup_only', '')
+        if reverse_lookup_only:
+            reverse_lookup_only = '<reverse_lookup_only>%s</reverse_lookup_only>' % reverse_lookup_only
+        
+        reverse_lookup_unify = kwargs.get('reverse_lookup_unify', '')
+        if reverse_lookup_unify:
+            reverse_lookup_unify = '<reverse_lookup_unify>%s</reverse_lookup_unify>' % reverse_lookup_unify
+
+        port_range = kwargs.get('port_range', '')
+        if port_range:
+            port_range = '<port_range>%s</port_range>' % port_range
+
+        port_list = kwargs.get('port_list', '')
+        if port_list:    
+            port_list = '<port_list id="%s"/>' % port_list
+
+        return '<modify_target target_id="{0}">{1}{2}{3}{4}{5}{6}  \
+                {7}{8}{9}{10}</modify_target>' \
+               ''.format(
+                         target_id, name, hosts, comment, copy, exclude_hosts,
+                         alive_tests, reverse_lookup_only, reverse_lookup_unify,
+                         port_range, port_list
+                         )
+
+    def modifyTaskCommand(self, task_id, kwargs):
+        
+        if not task_id:
+            raise ValueError('modify_task requires a task_id element')
+
+        name = kwargs.get('name', '')
+        if name:
+            name = '<name>%s</name>' % name
+
+        comment = kwargs.get('comment', '')
+        if comment:
+            comment = '<comment>%s</comment>' % comment
+
+        target_id = kwargs.get('target_id', '')
+        if target_id:
+            target_id = '<target id="%s"/>' % target_id
+
+        scanner = kwargs.get('scanner', '')
+        if scanner:
+            scanner = '<scanner_id="%s"></scanner>' % scanner
+            
+        schedule_periods = kwargs.get('schedule_periods', '')
+        if schedule_periods:
+            schedule_periods = '<schedule_periods>%d</schedule_periods>' % schedule_periods
+
+        schedule = kwargs.get('schedule', '')
+        if schedule:
+            schedule = '<schedule id="%s"/>' % (schedule)
+
+        alert = kwargs.get('alert', '')
+        if alert:
+            alert = '<alert id="%s"/>' % (alert)
+
+        observers = kwargs.get('observers', '')
+        if observers:
+            observers = '<observers>%s</observers>' % observers
+
+        preferences = kwargs.get('preferences', '')
```

Comment:
Could you give an example how the values for preferences are given to this function?

Reply:
As a dictionary where, preferences = {"scanner_name": ["name1", "name2"], "value": ["value1", "value2"]}.
perhaps there is a more efficient or desirable way of doing this?